pkg/vsphere/disk/disk_manager.go

```diff
@@ -421,3 +428,48 @@ func (m *Manager) devicePathByURI(op trace.Operation, datastoreURI *object.Datas
 
 	return sysPath, nil
 }
+
+func (m *Manager) InUse(op trace.Operation, config *VirtualDiskConfig) (*vm.VirtualMachine, error) {
```

Comment:
There are two primary use cases here:
1. find the _singluar_ VM that is powered on and using the disk (vSphere enforces this unless `ddb.multiWriter` is set to true)
2. find all VMs with the disk attached, including powered off VMs 

For this method to be used as a replacement for the exec cache lookup we can either have this method take a filter argument for power state or we can have it return a list and have that filtering done at a higher level.